### PR TITLE
To Do List - JS Project

### PR DESCRIPTION
**Pull request by AbdumurodovaZulfizar**
# ToDo List App

This Application will help you create your projects and your tasks on them every day. We used a webpack to sort JS files when building it. You can see JS ES6 features like closure, arrow function, eventListener and so on.

### **Learning Outcomes**

- Objects

- Factory Functions

- The Module Pattern

by Zulfizar Abdumurodova( Full-Stuck Software Developer)

Special thanks to all code-reviewers

P/S: While doing implemented changes I have to set up the project from scratch, [this](https://github.com/AbdumurodovaZulfizar/TodoList-JS/pull/1) is the link to the previous review.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,66 @@
...
+cd to-do-list-app
+```
+### Prerequisites
+
+- Node
+- Webpack
+
+### Setup
+- run `npm install` to install all required node modules
+- if you vant to see live version on your local machine run `npm run build` and open live server
```
Comment (rindrajosia) [inline README.md]: - [ ] {REQUIRED] Please use tools like grammarly in order to help you with your readme as it is the first thing users check. Here for example, chenge `vant` to `want`

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi , 

Good job so far!
There are some issues to revise before continuing to the next project but hang in there and read the comments below with care. You can do it! :muscle:


## Required Changes ♻️

- [x] It includes a rule for the `node_modules` and `.vscode` directory on `.gitignore` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/gitignore_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622602/))
- [x] I can just create one todo item for one project. Please fix it.
- [x] Keep all the styling logic in CSS files (If you need to change something like a background image, add the classes with the different backgrounds to the CSS and then change the class name attribute with JavaScript) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/styling_requirement.png) from Pathwright). As you didn't add `node_modules `to your` .gitignore`, I add a big difficulty to add inline comments to the right files.

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by AbdumurodovaZulfizar** [comment]
image
image


It is available to create several projects on both the live version and local host. I think something wrong with your tools.

**Comment by AbdumurodovaZulfizar** [comment]
I kept all logic on separate folders and I used the view files to show and style elements. It is also possible with Ruby apps

**Comment by rindrajosia** [comment]
> I kept all logic on separate folders and I used the view files to show and style elements. It is also possible with Ruby apps

I understand but in fact: all style like: display: none or display: block should be in a class in the css files then, you just use this class when the codition is reached. (As I told, you commit the node_module as a result, I had a big difficulty to add inline comment. Anyway, check my feedback)

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    controller.showTodos(parseInt(project, 10));
+  }
+
+  if (e.target.classList.contains('edit-todo')) {
+    const todo = e.target.getAttribute('data-todo');
+    const project = e.target.getAttribute('data-project');
+    const controller = todoController(todoModel, todoView);
+    controller.editTodo(parseInt(project, 10), parseInt(todo, 10));
+    document.querySelector('#add-todo').style.display = 'none';
+    document.querySelector('#update-todo').style.display = 'block';
```
Comment (bellom) [inline src/index.js]: Oops 🤦 Great work guys, your code and your design look really good, we just need to do more separation of the CSS logic. 🙏 just as the previous reviewer mentioned.

image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check my comments under this review;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best**

**Review by BrianSammit — APPROVED** [review]
STATUS APPROVED 🟢
--------
Hello @AbdumurodovaZulfizar, great job you did on this project so far 👍 💪 
CONGRATS!!1
---------
NOW YOU CAN MERGE THIS BRANCH 
HAPPY CODING!!!

Well-Done-Applause

**Pull request merged**